cognite/client/utils/_identifier.py

```diff
@@ -314,6 +317,16 @@ def load(
                 raise TypeError(
                     f"{id_name}external_ids must be of type str or SequenceNotStr[str]. Found {type(external_ids)}"
                 )
+        if instance_ids is not None:
+            if isinstance(instance_ids, InstanceId):
+                value_passed_as_primitive = True
+                all_identifiers.append(instance_ids)
+            elif isinstance(instance_ids, Sequence):
+                all_identifiers.extend(instance_ids)
```

Comment:
The other steps before this one enforce int/str before extending, so I think we should enforce `InstanceId` here